### PR TITLE
Fixed remarks and finished backend part

### PR DESCRIPTION
**Pull request by Mavlyudabegim**
In this PR I implemented authentication, registration logic using jwt, bcrypt
As well I added subscription logic
I fixed previous remarks

### CODE REVIEW

File: server/controllers/userController.js [hunk 1]
```diff
@@ -1,48 +1,83 @@
...
-    res.status(500).json(error);
+    next(error);
   }
 }
-async function user_create(req, res) {
+async function login(req, res, next) {
   try {
-    const user = await userService.createUser(req.body);
-    return res.status(201).json(user);
+    const userData = await userService.login(req.body.email, req.body.password);
```
Comment (Lokki17) [inline server/controllers/userController.js:25]: Why do you pass refresh token in cookies and as part of body?

File: server/controllers/userController.js [hunk 2]
```diff
@@ -1,48 +1,83 @@
...
+    next(error);
   }
 }
-async function users_details(req, res) {
+async function logout(req, res, next) {
   try {
-    const users = await userService.getAllUsers();
-    return res.status(200).json(users);
+    const { refreshToken } = req.cookies;
+    const token = await userService.logout(refreshToken);
```
Comment (Lokki17) [inline server/controllers/userController.js:38]: Why do you return token on logout?

File: server/controllers/userController.js [hunk 3]
```diff
@@ -1,48 +1,83 @@
...
+    res.clearCookie('refreshToken');
+    return res.status(200).json(token);
+  } catch (error) {
+    next(error);
+  }
+}
+
+async function refresh(req, res, next) {
```
Comment (Lokki17) [inline server/controllers/userController.js:46]: What does this method?

**Review by Lokki17 — CHANGES_REQUESTED** [review]
Fix remarks and merge

File: server/controllers/userController.js (hunk 1 above)
Comment (Mavlyudabegim) [reply server/controllers/userController.js:25]: I wanted to save refresh token in httpOnly cookie

File: server/controllers/userController.js (hunk 3 above)
Comment (Mavlyudabegim) [reply server/controllers/userController.js:46]: This methods gets request cookie and removes it from cookies.

File: server/controllers/userController.js (hunk 1 above)
Comment (Mavlyudabegim) [reply server/controllers/userController.js:25]: I wanted to save refresh token in httpOnly cookie

File: server/controllers/userController.js (hunk 3 above)
Comment (Mavlyudabegim) [reply server/controllers/userController.js:46]: This methods gets request cookie and removes it from cookies.

File: server/controllers/userController.js (hunk 2 above)
Comment (Mavlyudabegim) [reply server/controllers/userController.js:38]: On order to remove refreshToken from cookie I get refreshToken from httpOnly cookie

File: server/controllers/userController.js (hunk 1 above)
Comment (Lokki17) [reply server/controllers/userController.js:25]: ok, why?

File: server/controllers/userController.js (hunk 2 above)
Comment (Lokki17) [reply server/controllers/userController.js:38]: I dont understand. Why do you need to store token in cookie?

**Pull request merged**